### PR TITLE
fix: add audio padding

### PR DESCRIPTION
**Pull request by crystallee-ai**

### CODE REVIEW

File: hallo/utils/util.py [hunk 1]
```diff
@@ -291,7 +292,6 @@ def get_fps(video_path):
     container.close()
     return fps
 
-
```
Comment (siyuzhu-fudan) [inline hallo/utils/util.py:294]: Only submit necessary code changes.

File: hallo/utils/util.py [hunk 2]
```diff
@@ -303,19 +303,33 @@ def tensor_to_video(tensor, output_video_file, audio_source, fps=25):
...
-    ).numpy()  # convert to [f, h, w, c]
-    tensor = np.clip(tensor * 255, 0, 255).astype(
-        np.uint8
-    )  # to [0, 255]
+    ).numpy()  # Convert tensor to [f, h, w, c]
+    # Normalize to [0, 255]
+    tensor = np.clip(tensor * 255, 0, 255).astype(np.uint8)
 
     def make_frame(t):
-        # get index
```
Comment (siyuzhu-fudan) [inline hallo/utils/util.py:312]: Only submit necessary code changes.

File: scripts/inference.py [hunk 3]
```diff
@@ -280,7 +280,15 @@ def inference_process(args: argparse.Namespace):
...
+    if audio_emb.shape[0] % clip_length != 0:
+        padding_emb = torch.zeros(
+            (clip_length - audio_emb.shape[0] % clip_length,
+             audio_emb.shape[1],
+             audio_emb.shape[2],
+             audio_emb.shape[3]))
+        padding_emb = padding_emb.to(
+            dtype=audio_emb.dtype, device=audio_emb.device)
+        audio_emb = torch.cat([audio_emb, padding_emb], dim=0)
     times = audio_emb.shape[0] // clip_length
```
Comment (subazinga) [inline scripts/inference.py:292]: 建议用torch.nn.functional.pad函数做padding

Comment (subazinga) [reply scripts/inference.py:292]: 可能replicate padding会比zero padding合理一些吧？

**Review by subazinga — COMMENTED** [review]
We need to remove the frames generated by padding voice

**Comment by subazinga** [comment]
commit log should be squeezed

**Review by siyuzhu-fudan — APPROVED** [review]
